File: backend/queute/src/main/java/tn/cloudnine/queute/model/roadmap/FollowedRoadMap.java
package tn.cloudnine.queute.model.roadmap;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import tn.cloudnine.queute.model.user.User;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class FollowedRoadMap {
    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id ;
    private String frTitle;
    private String frDescription;
    @ManyToOne
    private User user;
    private Long roadMapId;
    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    private List<StepProgress>stepProgressList;
}

File: frontend/src/app/models/FollowedRoadMap.ts
import { StepProgress } from "./StepProgress";
import { User } from "./User";

export class FollowedRoadMap {
    id: number | null;
    frTitle:string;
    frDescription: string;
    roadMapId: number;  
    user:User;
    stepProgressList: StepProgress[];

    constructor(init?: Partial<FollowedRoadMap>) {
        this.id = init?.id ?? null;
        this.frTitle = init?.frTitle ?? "";
        this.frDescription = init?.frDescription ?? "";
        this.roadMapId = init?.roadMapId ?? 0;
        this.user = init?.user ?? new User();
        this.stepProgressList = (init?.stepProgressList ?? []).map(
            progress => new StepProgress(progress)
        );
    }
}